rust/src/dcerpc/dcerpc.rs

```diff
@@ -0,0 +1,620 @@
+        return DCERPCUDPResponse {
+            stub_data_buffer: Vec::new(),
+            stub_data_buffer_len: 0,
+        };
+    }
+}
+
+#[derive(Debug)]
+pub struct DCERPCUuidEntry {
+    pub ctxid: u16,
+    pub internal_id: u16,
+    pub result: u16,
+    pub uuid: Vec<u8>,
+    pub version: u16,
+    pub versionminor: u16,
+    pub flags: u16,
+}
+
+impl DCERPCUuidEntry {
+    pub fn new() -> DCERPCUuidEntry {
+        return DCERPCUuidEntry {
+            ctxid: 0,
+            internal_id: 0,
+            result: 0,
+            uuid: Vec::new(),
+            version: 0,
+            versionminor: 0,
+            flags: 0,
+        };
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Uuid {
+    pub time_low: Vec<u8>,
+    pub time_mid: Vec<u8>,
+    pub time_hi_and_version: Vec<u8>,
+    pub clock_seq_hi_and_reserved: u8,
+    pub clock_seq_low: u8,
+    pub node: Vec<u8>,
+}
+
+#[derive(Debug)]
+pub struct DCERPCHdrUdp {
+    pub rpc_vers: u8,
+    pub pkt_type: u8,
+    pub flags1: u8,
+    pub flags2: u8,
+    pub drep: Vec<u8>,
+    pub serial_hi: u8,
+    pub objectuuid: Vec<u8>,
+    pub interfaceuuid: Vec<u8>,
+    pub activityuuid: Vec<u8>,
+    pub server_boot: u32,
+    pub if_vers: u32,
+    pub seqnum: u32,
+    pub opnum: u16,
+    pub ihint: u16,
+    pub ahint: u16,
+    pub fraglen: u16,
+    pub fragnum: u16,
+    pub auth_proto: u8,
+    pub serial_lo: u8,
+}
+
+#[derive(Debug)]
+pub struct DCERPCUDPState {
+    pub dcerpchdrudp: Option<DCERPCHdrUdp>,
+    pub dcerpcudprequest: Option<DCERPCUDPRequest>,
+    pub dcerpcudpresponse: Option<DCERPCUDPResponse>,
+    pub fraglenleft: u16,
+    pub uuid_entry: Option<DCERPCUuidEntry>,
+    pub uuid_list: Vec<DCERPCUuidEntry>,
+    pub de_state: Option<*mut core::DetectEngineState>,
+}
+
+impl DCERPCUDPState {
+    pub fn new() -> DCERPCUDPState {
+        return DCERPCUDPState {
+            dcerpchdrudp: None,
+            dcerpcudprequest: None,
+            dcerpcudpresponse: None,
+            fraglenleft: 0,
+            uuid_entry: None,
+            uuid_list: Vec::new(),
+            de_state: None,
+        };
+    }
+
+    fn new_request(&mut self) {
+        let request = DCERPCUDPRequest::new();
+        self.dcerpcudprequest = Some(request);
+    }
+
+    fn new_response(&mut self) {
+        let response = DCERPCUDPResponse::new();
+        self.dcerpcudpresponse = Some(response);
+    }
+
+    fn create_new_tx(&mut self, pkt_type: u8) {
+        match pkt_type {
+            DCERPC_TYPE_REQUEST => {
+                self.new_request();
+            }
+            DCERPC_TYPE_RESPONSE => {
+                self.new_response();
+            }
+            _ => {
+                SCLogDebug!("Unrecognized packet type");
+            }
+        }
+    }
+
+    fn get_hdr_pkt_type(&self) -> u8 {
+        if let Some(ref hdr) = &self.dcerpchdrudp {
+            return hdr.pkt_type;
+        }
+        // Shouldn't happen
+        return 99;
+    }
```

Comment:
I think it would be more idiomatic to return Option or Result here - probably a Result if it should never really happen, or an option if it may, but unlikely to happen.

In the C binding we may want to return a value to designate error, but in Rust we should stick to the Rust idioms for new code.